hack/generator/spec/versioning.md

```diff
@@ -0,0 +1,192 @@
+# Versioning
+
+
+
+## Goals
+
+**Auto-generate conversions:** As far as practical, we want to autogenerate the schema for each hub version along with the conversions to and from the actual ARM API versions. Hand coding all the required conversions doesn't scale across all the different Azure sevices, especially with the ongoing rate of change. 
+
+**Allow for hand coded conversions:** While we expect to use code generation to handle the vast majority of needed conversions, we anticipate that some breaking API changes will require some of the conversion to be hand coded. We need to make it simple for these conversions to be introduced while still automatically generating the majority of the conversion.
+
+**No modification of generated files.** Manual modification of generated files is a known antipattern that greatly increases the complexity and burden of updates. If some files have been manually changed, every difference showing after code generation needs to be manually reviewed before being committed. This is tediuos and error prone because the vast majority of auto generated changes will be perfectly fine. 
+
+## Non-Goals
+
+**Coverage of every case by code generation:** While it's likely that very high coverage will be achievable with code generation, we don't believe that it will be practical to handle every possible situation automatically. It's therefore necessary for the solution to have some form of extensibility allowing for the injection of hand written code.
+
+## Proposed Solution
+
+The "v1" storage version of each supported resource type will be created by merging all of the fields of all the distinct versions of the resource type, creating a *superset* type that includes every property declared across every version of the API.
+
+* To maintain backward compatibility as Azure APIs evolve over time, we will include properties across all versions of the API, even for versions we are not currently generating as output. This ensures that properties in use by older APIs are still present and available for forward conversion to newer APIs, even as those older APIs age out of use.
```

Comment:
The argument for avoiding accumulation of "storage schema debt" is significant; I've rewritten the proposal to avoid this.

Reply:
The proposal now recommends a rolling-storage version to avoid the unbounded accumulation of storage schema debt over time. It's more complex to set up initially, but doesn't degrade over time.